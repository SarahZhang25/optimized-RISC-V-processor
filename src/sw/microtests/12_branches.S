// This is a test that tests that your processor handles all the BRANCH
// instructions correctly.
.text
start:
    li x1, 1
    li x2, 2
    li x3, -1
    bne x1, x1, wrong
    li x4, 1
    beq x1, x2, wrong
    li x5, 1
    blt x1, x1, wrong
    li x6, 1
    blt x2, x1, wrong
    li x7, 1
    blt x1, x3, wrong
    li x8, 1
    bge x1, x2, wrong
    li x9, 1
    bge x3, x1, wrong
    li x10, 1
    bltu x1, x1, wrong
    li x11, 1
    bltu x2, x1, wrong
    li x12, 1
    bltu x3, x2, wrong
    li x13, 1
    bgeu x1, x2, wrong
    li x14, 1
    bgeu x2, x3, wrong
    li x15, 1

    beq x1, x1, a
    unimp
a:  li x16, 1
    bne x1, x2, b
    unimp
b:  li x17, 1
    blt x1, x2, c
    unimp
c:  li x18, 1
    blt x3, x1, d
    unimp
d:  li x19, 1
    bge x2, x1, e
    unimp
e:  li x20, 1
    bge x1, x1, f
    unimp
f:  li x21, 1
    bge x1, x3, g
    unimp
g:  li x22, 1
    bltu x1, x2, h
    unimp
h:  li x23, 1
    bltu x2, x3, i
    unimp
i:  li x24, 1
    bgeu x1, x1, j
    unimp
j:  li x25, 1
    bgeu x2, x1, k
    unimp
k:  li x26, 1
    bgeu x3, x2, done
    unimp

done:
    li x27, 1
    unimp

wrong:
    unimp
