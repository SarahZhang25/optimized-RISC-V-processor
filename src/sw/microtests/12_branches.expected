Dumping the state of the processor
pc = 0x000000fc
x0 = 0x00000000
x1 = 0x00000001
x2 = 0x00000002
x3 = 0xffffffff
x4 = 0x00000001
x5 = 0x00000001
x6 = 0x00000001
x7 = 0x00000001
x8 = 0x00000001
x9 = 0x00000001
x10 = 0x00000001
x11 = 0x00000001
x12 = 0x00000001
x13 = 0x00000001
x14 = 0x00000001
x15 = 0x00000001
x16 = 0x00000001
x17 = 0x00000001
x18 = 0x00000001
x19 = 0x00000001
x20 = 0x00000001
x21 = 0x00000001
x22 = 0x00000001
x23 = 0x00000001
x24 = 0x00000001
x25 = 0x00000001
x26 = 0x00000001
x27 = 0x00000001
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
